src/lib/enquirer.js

```diff
@@ -0,0 +1,58 @@
+const Enquirer = require('enquirer');
+const chalk = require('chalk');
+
+function disablePromptsInCI(answers = {}) {
+  return enquirer => {
+    const prompt = enquirer.prompt.bind(enquirer);
+    const context = { ...enquirer.answers, ...answers };
+
+    // eslint-disable-next-line no-param-reassign
+    enquirer.prompt = async questions => {
+      const list = [].concat(questions || []);
+
+      // Purposefully loading `ci-info` here so we can adequately test it.
+      const ci = require('ci-info'); // eslint-disable-line global-require
+      if (!ci.isCI) {
+        // If we aren't in a CI environment then we shouldn't look for unanswered questions.
+        return prompt(list);
+      }
+
+      const missingArgs = [];
+
+      (questions || []).forEach(item => {
+        // Confirm-style questions are always optional!
+        if (item.type !== 'confirm') {
+          const value = context[item.name];
+          if (value === undefined && !item.skip()) {
+            missingArgs.push(item.name);
+          }
+        }
+      });
+
+      if (missingArgs.length) {
+        throw new Error(
+          `We've detected you're running within a ${
+            ci.name
+          } environment, please supply the following required arguments for your command instead of relying on our terminal prompt system: ${chalk.yellow(
+            missingArgs.join(',')
+          )}`
+        );
+      }
+
+      return prompt(list);
+    };
+  };
+}
+
+module.exports = function (answers = {}, questions = []) {
+  const enquirer = new Enquirer();
+
+  enquirer.use(disablePromptsInCI(answers));
+
+  return enquirer.prompt(questions).catch(err => {
+    // If the Enquirer promise was rejected without an error then the user killed an open prompt so
+    // we should abide by their wishes and bail out.
+    if (!err) process.exit(0);
```

Comment:
I feel like we could return a `1` here since it's technically not a success, right? Not sure what the best practices are here 🤔

Reply:
Exit code `1` generally means that that app executed with errors but since in this case I'm killing the prompt myself it's not really an error. Jest also returns the same `0` exit code when you exit out of an open `--watch`.

Exiting Jest before it's able to load any tests:

![Screen Shot 2022-02-27 at 7 39 07 PM](https://user-images.githubusercontent.com/33762/155920130-d5d2ca3c-31b3-4ad7-9969-2b2b77915963.png)

Exiting an open `--watch` process:

![Screen Shot 2022-02-27 at 7 39 15 PM](https://user-images.githubusercontent.com/33762/155920143-9f3bf026-77ed-429f-9c2b-e1789c0336d3.png)